lib/ansible/module_utils/basic.py

```diff
@@ -2134,6 +2134,11 @@ def fail_json(self, **kwargs):
         if 'changed' not in kwargs:
             kwargs['changed'] = False
 
+        # add traceback if debug or high verbosity and it is missing
+        # note: badly named as exception, it is really always been 'traceback'
+        if 'exception' not in kwargs and (self._debug or self._verbosity >= 3):
```

Comment:
I'd add a check that there's a traceback being processed:
``` python
if 'exception' not in kwargs and sys.exc_info()[2] and (self._debug or self._verbosity >= 3):
```

That way the handling of results['exception'] in controller-side code will be easier.